### ISSUE TITLE
Spyder update process improvements & bugs

### ISSUE THREAD

**Issue by Caseyb87** [spyder-ide/spyder#17424]
Updating stand-alone Spyder in Windows leaves a lot to be improved. Having just updated from 5.2.1 to 5.2.2 I have the following bug report and suggestions.

**Suggestion 1:** Updates should be handled internally in Spyder and not require the user to download and execute the new version's .exe file (i.e. Spyder should launch an "updater" process to do the work behind the scenes).

**Bug:** when attempting to install an updated version, the installer recognises that a previous version of Spyder is installed and says this will need to be removed. OK (though see suggestion 2). After clicking "Yes" to remove the existing version, the installer removes the old version of Spyder but then finishes on a screen where the only option is to "Close" the installer. The installer does not install the latest version. The user has to the restart the installer for it to install the new version of Spyder. 

**Suggestion 2:** fully uninstalling the previous version is a bit of an annoyance. It removes things like shortcuts and just seems like a bit of a waste of time/resource - as presumably not everything needs to be deleted and recreated. Could the installer actually act as an updater and only modify files that need to be updated?

**Comment by ccordoba12** [comment]
Hey @Caseyb87, thanks for reporting. We are aware of these limitations and right now we're looking for funding to improve precisely the shortcomings you mentioned above.

**Comment by ERF4** [comment]
Yes, just throwing in my 2 cents that this is a really needed feature. Best of luck securing the funding!
